src/mongo/shell.ts

```diff
@@ -98,6 +100,16 @@ export class Shell {
 				5000);
 			const disposable = this.onResult.event(result => {
 				disposable.dispose();
+
+				if (result.code) {
+					if (result.code === 'ENOENT') {
+						result.message = `Could not find Mongo shell. Make sure it is on your path or you have set the '${ext.settingsKeys.mongoShellPath}' VS Code setting to point to the Mongo shell executable file. Attempted command: "${this.execPath}"`;
+					}
+
+					e(result);
```

Comment:
Do you still need to call `clearTimeout(handler);` at some point?

Reply:
Fixed